numpy/f2py/service.py

```diff
@@ -0,0 +1,327 @@
+        sys.stdout.write(pyf)
+    else:
+        with open(sign_file, "w") as f:
+            f.write(pyf)
+
+def _check_postlist(postlist, sign_file: str):
+    isusedby = {}
+    for plist in postlist:
+        if 'use' in plist:
+            for u in plist['use'].keys():
+                if u not in isusedby:
+                    isusedby[u] = []
+                isusedby[u].append(plist['name'])
+    for plist in postlist:
+        if plist['block'] == 'python module' and '__user__' in plist['name'] and plist['name'] in isusedby:
+            # if not quiet:
+            outmess(
+                f'Skipping Makefile build for module "{plist["name"]}" '
+                'which is used by {}\n'.format(
+                    ','.join(f'"{s}"' for s in isusedby[plist['name']])))
+    if(sign_file):
+        outmess(
+            'Stopping. Edit the signature file and then run f2py on the signature file: ')
+        outmess('%s %s\n' %
+                (os.path.basename(sys.argv[0]), sign_file))
+        return
+    for plist in postlist:
+        if plist['block'] != 'python module':
+            # if 'python module' not in options:
+                outmess(
+                    'Tip: If your original code is Fortran source then you must use -m option.\n')
+            # raise TypeError('All blocks must be python module blocks but got %s' % (
+            #     repr(postlist[i]['block'])))
+
+def _callcrackfortran(files: List[Path], module_name: str):
+    postlist = crackfortran.crackfortran([str(file) for file in files])
+    for mod in postlist:
+        mod["coutput"] = f"{module_name}module.c"
+        mod["f2py_wrapper_output"] = f"{module_name}-f2pywrappers.f"
+    return postlist
+
+def _set_dependencies_dist(ext_args, link_resource):
+    for dep in link_resource:
+        info = get_info(dep)
+        if not info:
+                outmess('No %s resources found in system'
+                        ' (try `f2py --help-link`)\n' % (repr(n)))
+        dict_append(ext_args, **link_resource)
+
+def get_f2py_modulename(source):
+    name = None
+    _f2py_module_name_match = re.compile(r'\s*python\s*module\s*(?P<name>[\w_]+)',
+                                        re.I).match
+    _f2py_user_module_name_match = re.compile(r'\s*python\s*module\s*(?P<name>[\w_]*?'
+                                            r'__user__[\w_]*)', re.I).match
+    with open(source) as f:
+        for line in f:
+            if m := _f2py_module_name_match(line):
+                if _f2py_user_module_name_match(line): # skip *__user__* names
+                    continue
+                name = m.group('name')
+                break
+    return name
+
+def wrapper_settings(rules_setts, crackfortran_setts, capi_maps_setts, auxfuncs_setts):
+    _set_rules(rules_setts)
+    _set_crackfortran(crackfortran_setts)
+    _set_capi_maps(capi_maps_setts)
+    _set_auxfuncs(auxfuncs_setts)
+
+def generate_files(files: List[Path], module_name: str, sign_file: str):
```

Comment:
I'm not hugely fond of typing, but if we have it here we might want it for all the functions?

Reply:
Yep, as suggested by @BvB93, I am changing the typing to builtins and will add it for `f2pyarg.py` and `service.py`